code/datums/martial/worldshaker.dm

```diff
@@ -0,0 +1,546 @@
+	return TRUE //you're doing enough pushing as is
+
+/datum/martial_art/worldshaker/harm_act(mob/living/carbon/human/A, mob/living/D)
+	return TRUE //no punch, just pummel
+
+/datum/martial_art/worldshaker/proc/InterceptClickOn(mob/living/carbon/human/H, params, atom/target)
+	var/list/modifiers = params2list(params)
+	if(!(can_use(H)) || (modifiers["shift"] || modifiers["alt"] || modifiers["ctrl"]))
+		return
+	if(H.a_intent == INTENT_DISARM)
+		leap(H, target)
+	if(H.a_intent == INTENT_HELP && (H==target))
+		rip_plate(H)
+	if(thrown.len > 0 && H.a_intent == INTENT_GRAB)
+		if(get_turf(target) != get_turf(H))
+			lob(H, target)
+
+	if(!H.Adjacent(target) || H==target)
+		return
+	if(H.a_intent == INTENT_HARM && isliving(target))
+		pummel(H,target)
+	if(H.a_intent == INTENT_GRAB)
+		grapple(H,target)
+
+	
+/*-------------------------------------------------------------
+	start of helpers section
+---------------------------------------------------------*/
+/datum/martial_art/worldshaker/proc/stagger(mob/living/victim)
+	if(HAS_TRAIT(victim, TRAIT_STUNIMMUNE))
+		return
+	victim.set_resting(TRUE)//basically a trip
+	victim.add_movespeed_modifier(id, update=TRUE, priority=101, multiplicative_slowdown = 0.5)
+	addtimer(CALLBACK(src, PROC_REF(stagger_end), victim), STAGGER_DURATION, TIMER_UNIQUE | TIMER_OVERRIDE)
+
+/datum/martial_art/worldshaker/proc/stagger_end(mob/living/victim)
+	victim.remove_movespeed_modifier(id)
+
+/datum/martial_art/worldshaker/proc/push_away(mob/living/user, atom/movable/victim, distance = 1)
+	var/throwdirection = get_dir(user, victim)
+	var/atom/throw_target = get_edge_target_turf(victim, throwdirection)
+	var/throwspeed = 3
+	if(heavy)
+		throwspeed *= 2
+		distance *= 2
+	victim.throw_at(throw_target, distance, throwspeed, user)
+/*---------------------------------------------------------------
+	end of helpers section
+----------------------------------------------------------------*/
+/*---------------------------------------------------------------
+	start of plates section 
+---------------------------------------------------------------*/
+/datum/martial_art/worldshaker/proc/grow_plate(mob/living/carbon/human/user)
+	if(plates >= PLATE_CAP || user.stat == DEAD)//no quaking the entire station
+		return
+	user.balloon_alert(user, span_notice("Your plates grow thicker!"))
+	user.adjustBruteLoss(-2)//more for flavour than actual gameplay (and so the damage from tearing off a plate isn't annoying)
+	user.adjustFireLoss(-1)
+	plates++
+	if(plates <= MAX_PLATES)
+		user.physiology.damage_resistance += PLATE_REDUCTION
+	update_platespeed(user)
+
+/datum/martial_art/worldshaker/proc/rip_plate(mob/living/carbon/human/user)
+	if(user.get_active_held_item())
+		if(COOLDOWN_FINISHED(src, next_balloon))
+			COOLDOWN_START(src, next_balloon, BALLOON_COOLDOWN)
+			user.balloon_alert(user, span_warning("You need an empty hand to tear off some of your plate!"))
```

Comment:
i made the comment on the upper one but this is the one i think is a bit of a problem because of size and whatnot